packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -20,23 +20,36 @@ impl TryFrom<&[Attribute]> for Value {
             .iter()
             .filter(|attr| attr.path().is_ident("status") || attr.path().is_ident("value"))
             .exactly_one()
-            .map_err(|_| {
-                "exactly one of #[status] and #[value(Type)] must be provided".to_string()
-            })?;
-
-        if let Meta::List(list) = &attr.meta {
-            if attr.path().is_ident("value") {
-                Ok(Self::Type(
-                    list.parse_args::<Type>()
-                        .map_err(|_| "failed to parse value type".to_string())?,
-                ))
-            } else {
-                Err("status attribute cannot have parameters".to_string())
-            }
-        } else if attr.path().is_ident("status") {
+            .map_err(|_| "exactly one of #[status] and #[value(Type)] must be provided")?;
+
+        if attr.path().is_ident("status") {
             Ok(Self::Status)
+        } else if let Meta::List(list) = &attr.meta {
+            Ok(Self::Type(
+                list.parse_args::<Type>()
+                    .map_err(|_| "failed to parse value type")?,
+            ))
         } else {
-            Err("value attribute must contain a type parameter: #[value(Type)]".to_string())
+            Err("value attribute must contain a type parameter: #[value(Type)]".into())
+        }
+    }
+}
+
+trait FieldsExt {
+    fn fields_data(&self) -> (Vec<&Option<Ident>>, Vec<&Type>);
```

Comment:
```suggestion
    fn names(&self) -> Vec<&Option<Ident>>;

    fn types(&self) -> Vec<&Type>;
```
decouple independent return values. This avoids the `(.., _) = self.fields.fields_data()` oddity